### DISCUSSION TITLE
Prevent Metals from locking the files (VSCode)

### DISCUSSION THREAD

**Discussion by gsavini-mce** [scalameta/metals#5686] [Q&A] [answered]
In VSCode with the Metals extension, if I execute `maven install -e` with the same snapshot version while another maven project has the first project as a dependency, I get the following error:

`java.nio.file.AccessDeniedException: C:\[...]\.m2\repository\[...]\test\1.1-SNAPSHOT\test-1.1-SNAPSHOT.jar.7921050410306119960.tmp -> C:\[...]\.m2\repository\[...]\test\1.1-SNAPSHOT\test-1.1-SNAPSHOT.jar
`

It seems like that the file is locked by metals. I have tried without it and the file isn't locked. To solve this I have to quit VSCode or terminate the process.

Any suggestion to solve this problem?

**Comment by tgodzik** [comment, marked as answer]
I think this might be caused by Metals using those jar in the Classloader, which we use for example to find the modules used by a particular jar. We would need to rework a bunch of features to make it work without the classloader :thinking:
